events/sinks/influxdb/influxdb.go

```diff
@@ -0,0 +1,190 @@
+	bytes, err := json.MarshalIndent(event, "", " ")
+	if err != nil {
+		return "", err
+	}
+	return string(bytes), nil
+}
+
+func eventToPoint(event *kube_api.Event) (*influxdb.Point, error) {
+	value, err := getEventValue(event)
+	if err != nil {
+		return nil, err
+	}
+
+	point := influxdb.Point{
+		Measurement: eventMeasurementName,
+		Time:        event.LastTimestamp.Time.UTC(),
+		Fields: map[string]interface{}{
+			valueField: value,
+		},
+		Tags: map[string]string{
+			eventUID: string(event.UID),
+		},
+	}
+	if event.InvolvedObject.Kind == "Pod" {
+		point.Tags[metrics_core.LabelPodId.Key] = string(event.InvolvedObject.UID)
+		point.Tags[metrics_core.LabelPodName.Key] = event.InvolvedObject.Name
+	}
+	point.Tags[metrics_core.LabelHostname.Key] = event.Source.Host
+	return &point, nil
+}
+
+func (sink *influxdbSink) ExportEvents(eventBatch *core.EventBatch) {
+	sink.Lock()
+	defer sink.Unlock()
+
+	dataPoints := make([]influxdb.Point, 0, 10)
+	for _, event := range eventBatch.Events {
+		point, err := eventToPoint(event)
+		if err != nil {
+			glog.Warningf("Failed to convert event to point: %v", err)
+		}
+		dataPoints = append(dataPoints, *point)
+		if len(dataPoints) >= maxSendBatchSize {
+			sink.sendData(dataPoints)
+			dataPoints = make([]influxdb.Point, 0, 1)
+		}
+	}
+	if len(dataPoints) >= 0 {
+		sink.sendData(dataPoints)
+	}
+}
+
+func (sink *influxdbSink) sendData(dataPoints []influxdb.Point) {
+	if err := sink.createDatabase(); err != nil {
+		glog.Errorf("Failed to create infuxdb: %v", err)
+	}
+	bp := influxdb.BatchPoints{
+		Points:          dataPoints,
+		Database:        sink.c.DbName,
+		RetentionPolicy: "default",
+	}
+
+	start := time.Now()
+	if _, err := sink.client.Write(bp); err != nil {
+		if strings.Contains(err.Error(), dbNotFoundError) {
+			sink.resetConnection()
+		} else if _, _, err := sink.client.Ping(); err != nil {
+			glog.Errorf("InfluxDB ping failed: %v", err)
+			sink.resetConnection()
+		}
+	}
+	end := time.Now()
+	glog.V(4).Info("Exported %d data to influxDB in %s", len(dataPoints), end.Sub(start))
+}
+
+func (sink *influxdbSink) Name() string {
+	return "InfluxDB Sink"
+}
+
+func (sink *influxdbSink) Stop() {
+	// nothing needs to be done.
+}
+
+func (sink *influxdbSink) createDatabase() error {
+	if sink.client == nil {
+		client, err := influxdb_common.NewClient(sink.c)
+		if err != nil {
+			return err
+		}
+		sink.client = client
+	}
+
+	if sink.dbExists {
+		return nil
+	}
+	q := influxdb.Query{
+		Command: fmt.Sprintf("CREATE DATABASE %s", sink.c.DbName),
+	}
+	if resp, err := sink.client.Query(q); err != nil {
+		if !(resp != nil && resp.Err != nil && strings.Contains(resp.Err.Error(), "already exists")) {
+			return fmt.Errorf("Database creation failed: %v", err)
+		}
+	}
+	sink.dbExists = true
+	glog.Infof("Created database %q on influxDB server at %q", sink.c.DbName, sink.c.Host)
+	return nil
+}
+
+// Returns a thread-compatible implementation of influxdb interactions.
```

Comment:
What does it mean `influxdb interactions`? Does `interactions` have some special meaning in heapster?